sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -408,6 +408,14 @@ case class GpuCast(
       case (_: DecimalType, StringType) =>
         input.castTo(DType.STRING)
 
+      case (ArrayType(_@FloatType, _), ArrayType(nestedTo@DoubleType, _)) =>
+        val rapidsType = GpuColumnVector.getNonNestedRapidsType(nestedTo)
+        withResource(input.getChildColumnView(0)) { childView =>
+          withResource(childView.castTo(rapidsType)) { castChildView =>
```

Comment:
nit: castTo seems to return a ColumnVector. But it's also fine to match childView with castChildView.